apps/src/studio/Item.js

```diff
@@ -358,3 +444,37 @@ Item.prototype.moveToNextPosition = function () {
   this.x = next.x;
   this.y = next.y;
 };
+
+/**
+ * Mark that we're colliding with object represented by key.
+ * Here, override base implemention to special on-collision logic for certain
+ * item classes.
+ * @param key A unique key representing the object we're colliding with
+ * @returns {boolean} True if collision is started, false if we're already colliding
+ * @override
+ */
+Item.prototype.startCollision = function (key) {
+  var newCollisionStarted = Item.superPrototype.startCollision.call(this, key);
+  if (newCollisionStarted) {
+    if (this.className === 'hazard') {
+      var actor = Studio.sprite[key];
+      if (actor) {
```

Comment:
Requiring all controlling logic to live in the controller leads to a god-class; in terms of breaking a "game" down into organized bits, it makes sense to let each item class own and run its own effect.  That said, I'd much prefer that this happened in a non-global way with the item getting passed a reference to the collidable that it manipulates instead of this `key` (which can already mean multiple things) and going out to Studio to get the collidable itself.